src/periodic.jl

```diff
@@ -0,0 +1,85 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicParameter = Union{Real,Period}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::PeriodicFunction`: the periodic function
```

Comment:
```suggestion
* `f::Union{typeof(cos), typeof(sin)}`: the periodic function
```

else interpolate the type in the docs using `@docs` but this is probably easier